src/Storages/System/StorageSystemDDLWorkerQueue.cpp

```diff
@@ -0,0 +1,289 @@
+{
+    return std::vector<std::pair<String, Int8>>{
+        {"ZOK", static_cast<Int8>(Coordination::Error::ZOK)},
+        {"ZSYSTEMERROR", static_cast<Int8>(Coordination::Error::ZSYSTEMERROR)},
+        {"ZRUNTIMEINCONSISTENCY", static_cast<Int8>(Coordination::Error::ZRUNTIMEINCONSISTENCY)},
+        {"ZDATAINCONSISTENCY", static_cast<Int8>(Coordination::Error::ZDATAINCONSISTENCY)},
+        {"ZCONNECTIONLOSS", static_cast<Int8>(Coordination::Error::ZCONNECTIONLOSS)},
+        {"ZMARSHALLINGERROR", static_cast<Int8>(Coordination::Error::ZMARSHALLINGERROR)},
+        {"ZUNIMPLEMENTED", static_cast<Int8>(Coordination::Error::ZUNIMPLEMENTED)},
+        {"ZOPERATIONTIMEOUT", static_cast<Int8>(Coordination::Error::ZOPERATIONTIMEOUT)},
+        {"ZBADARGUMENTS", static_cast<Int8>(Coordination::Error::ZBADARGUMENTS)},
+        {"ZINVALIDSTATE", static_cast<Int8>(Coordination::Error::ZINVALIDSTATE)},
+        {"ZAPIERROR", static_cast<Int8>(Coordination::Error::ZAPIERROR)},
+        {"ZNONODE", static_cast<Int8>(Coordination::Error::ZNONODE)},
+        {"ZNOAUTH", static_cast<Int8>(Coordination::Error::ZNOAUTH)},
+        {"ZBADVERSION", static_cast<Int8>(Coordination::Error::ZBADVERSION)},
+        {"ZNOCHILDRENFOREPHEMERALS", static_cast<Int8>(Coordination::Error::ZNOCHILDRENFOREPHEMERALS)},
+        {"ZNODEEXISTS", static_cast<Int8>(Coordination::Error::ZNODEEXISTS)},
+        {"ZNOTEMPTY", static_cast<Int8>(Coordination::Error::ZNOTEMPTY)},
+        {"ZSESSIONEXPIRED", static_cast<Int8>(Coordination::Error::ZSESSIONEXPIRED)},
+        {"ZINVALIDCALLBACK", static_cast<Int8>(Coordination::Error::ZINVALIDCALLBACK)},
+        {"ZINVALIDACL", static_cast<Int8>(Coordination::Error::ZINVALIDACL)},
+        {"ZAUTHFAILED", static_cast<Int8>(Coordination::Error::ZAUTHFAILED)},
+        {"ZCLOSING", static_cast<Int8>(Coordination::Error::ZCLOSING)},
+        {"ZNOTHING", static_cast<Int8>(Coordination::Error::ZNOTHING)},
+        {"ZSESSIONMOVED", static_cast<Int8>(Coordination::Error::ZSESSIONMOVED)},
+    };
+}
+
+
+NamesAndTypesList StorageSystemDDLWorkerQueue::getNamesAndTypes()
+{
+    return {
+        {"entry", std::make_shared<DataTypeString>()},
+        {"host_name", std::make_shared<DataTypeString>()},
+        {"host_address", std::make_shared<DataTypeString>()},
+        {"port", std::make_shared<DataTypeUInt16>()},
+        {"status", std::make_shared<DataTypeEnum8>(getStatusEnumsAndValues())},
+        {"cluster", std::make_shared<DataTypeString>()},
+        {"value", std::make_shared<DataTypeString>()},
+        {"query_start_time", std::make_shared<DataTypeDateTime>()},
+        {"query_finish_time", std::make_shared<DataTypeDateTime>()},
+        {"query_duration_ms", std::make_shared<DataTypeUInt64>()},
+        {"exception_code", std::make_shared<DataTypeEnum8>(getZooKeeperErrorEnumsAndValues())},
+    };
+}
+
+
+static bool extractPathImpl(const IAST & elem, String & res, const Context & context)
```

Comment:
Looks like copy-paste from `StorageSystemZooKeeper.cpp`

Reply:
Yes, it is. I was going to leave a TODO to extract this function common to both files  (they currently use)in a follow up PR. Those functions are private to the `StorageSystemZookeeper.cpp` and also use a different ident `path` instead of `cluster`. I can make those changes in the same PR or can follow up.

Reply:
ended up removing this code as it's no longer needed.